server/identification.ts

```diff
@@ -118,7 +118,13 @@ class Identification {
 
 		this.connections.forEach((connection) => {
 			if (!connection.socket.remotePort || !connection.socket.localPort) {
-				throw new Error("Socket has no remote or local port");
+				// Race condition: this can happen when more than one socket gets disconnected at
+				// once, since we `refresh()` for each one being added/removed. This results
+				// in there possibly being one or more disconnected sockets remaining when we get here.
+				//
+				// Simply skip this socket and do not crash the server.
```

Comment:
The only reason it crashed is because we had a `throw` here, so this comment isn't really relevant with the log.

I hope the warning won't persist every time it refreshes, which would indicate that the socket is never removed. I'd add `id` to the log so if that does happen it could be caught.

Reply:
Makes sense, updated the PR